tests/cloner_test.go

```diff
@@ -531,6 +531,65 @@ var _ = Describe("all clone tests", func() {
 			})
 			ClonerBehavior(cloneStorageClassName, "csivolumeclone")
 		})
+
+		Context("CloneStrategy on storageclass annotation", func() {
+			cloneType := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyCsiClone)
+
+			BeforeEach(func() {
+				if !f.IsCSIVolumeCloneStorageClassAvailable() {
+					Skip("CSI Volume Clone is not applicable")
+				}
+				cloneStorageClassName = f.CsiCloneSCName
+				By(fmt.Sprintf("Get original storage profile: %s", cloneStorageClassName))
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				//TODO:  get storageClass `cloneStorageClassName` and annotate with `cloneType`
+				storageProfile.Spec.CloneStrategy = &cloneType
+
+			})
+			AfterEach(func() {
+				By("[AfterEach] Restore the storage class - remove annotation ")
+				// TODO: Remove annotation from storageClass `cloneStorageClassName`
```

Comment:
and here the same, no need to edit storageProfile, just remove annotation from storageClass and check if storageProfile was updated in an `Eventually()` loop

Reply:
Yeah basically do the reverse of what I showed for the BeforeEach, remove the annotation, and wait until the storage profile no longer has the Status.CloneStrategy. We are not setting the clone strategy in the Spec, so we don't care about that.